staging/src/k8s.io/apiserver/pkg/storage/storagebackend/factory/etcd3.go

```diff
@@ -145,13 +146,17 @@ func newETCD3Client(c storagebackend.TransportConfig) (*clientv3.Client, error)
 		}
 		dialOptions = append(dialOptions, grpc.WithContextDialer(dialer))
 	}
+	lcfg := logutil.DefaultZapLoggerConfig
+	// Disable sampling to save memory.
+	lcfg.Sampling = nil
```

Comment:
Do we need to configure the logging level here given that we are overwriting the log config? For reference:
https://github.com/etcd-io/etcd/blob/6c72c1b09cbf9a4950b4196f708028086be7fa68/client/v3/logger.go#L63:6

Reply:
In the current version of client in k8s (dd1b699fc4895de8cc23c3cac5a428c37eee384a) this code looks diffently:

https://github.com/etcd-io/etcd/blob/dd1b699fc4895de8cc23c3cac5a428c37eee384a/clientv3/client.go#L425-L428

but I see that https://github.com/kubernetes/kubernetes/pull/102061 updates that library to 6c72c1b09cbf9a4950b4196f708028086be7fa68 where this logic looks differently:

https://github.com/etcd-io/etcd/blob/6c72c1b09cbf9a4950b4196f708028086be7fa68/client/v3/client.go#L354-L360

I'm not sure how to proceed with this -- I would prefer to get a copy of existing log config (e.g. to keep ETCD_CLIENT_DEBUG env variable working) and override just a single field.

Can we make the  createDefaultZapLogger function public? Or make the log config adjustable in some other way?

Reply:
Yeah I think it makes sense to make createDefaultZapLogger public. But let's not be blocked on that.

Reply:
I think we can make CreateDefaultZapLoggerConfig public. 
I assume you care about the 'config' version that will let you override other properties.

Reply:
Proposed PR: https://github.com/etcd-io/etcd/pull/13039 in etcd